src/backgrounds.jl

```diff
@@ -0,0 +1,212 @@
+"""
+
+`draw_grid(video::Video, action::Action, frame::Int; grid_direction::Symbol = :TR)`
+
+Draws an oriented grid on the given frame of a Video.
+
+# Arguments
+- `grid_direction::Symbol`: Where grid animation finishes. Default: `:TR` Available Orientations:
+  - `:TR` - Animation finishes in the **T**op **R**ight corner of the frame.
+  - `:TL` - Animation finishes in the **T**op **L**eft corner of the frame.
+  - `:BR` - Animation finishes in the **B**ottom **R**ight corner of the frame.
+  - `:BL` - Animation finishes in the **B**ottom **L**eft corner of the frame.
+
+# Example
+This example will produce a grid animation that finishes in the top left corner of the animation. One will need to define their own path for `tempdirectory` and `pathname`.
+
+```
+using Javis
+using Luxor
+
+function ground(args...)
+    background("white")
+    sethue("black")
+end
+
+video = Video(500, 500)
+javis(
+      video,
+      [
+          Action(1:100, ground),
+          Action(1:100, (args...) -> draw_grid(args..., grid_direction = :TL)),
+      ],
+      tempdirectory = "/tmp-directory",
+      creategif = true,
+      pathname = "/tmp-directory/grid_animation.gif"
+     )
+
+```
+
+"""
+function draw_grid(video::Video, action::Action, frame::Int; grid_direction::Symbol = :TR)
+
+    min_width = div(video.width, -2, RoundDown)
+    max_width = div(video.width, 2, RoundUp)
+
+    min_height = div(video.height, -2, RoundDown)
+    max_height = div(video.height, 2, RoundUp)
+
+    if grid_direction == :BR
+
+        for y_point = min_width:25:max_width
+            step = (frame - first(action.frames)) / (length(action.frames) - 1)
+            start_point = Point(min_width, y_point)
+            end_point = start_point + step * (Point(max_width, y_point) - start_point)
+            line(start_point, end_point, :stroke)
```

Comment:
Same as with `step` but you want to have it at the very end.

Reply:
Fixed.

Reply:
`line` should be after the `if` as you repeat it 8 times.

Reply:
With the way it is right now, the for loop creates a horizontal line object each iteration in such a way that `start_point` and `end_point` change between each loop. If I were to put the line outside the if, each line would not be made. Am I missing something somewhere?

Reply:
No that's correct. I missed that or rather was wondering to make the loop outside of the if but that is computationally probably not the right thing to do.